### PR TITLE
adjust rbac to match mcad

### PR DESCRIPTION
**Pull request by jbusche**
# Issue link
Closes https://github.com/project-codeflare/codeflare-operator/issues/187

# What changes have been made
Adjusts the mcad_controller.go rbac rules to match the recent changes with MCAD in PR https://github.com/project-codeflare/multi-cluster-app-dispatcher/pull/475

# Verification steps
TBD

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by jbusche** [comment]
OK - i built a new operator image and tried deploying it:
```
oc get pod -n openshift-operators |grep codeflare
codeflare-operator-manager-59b49cf5b7-bdx2d                1/1     Running   1 (52m ago)   64m
```
Using my test image:
```
oc describe pod codeflare-operator-manager-59b49cf5b7-bdx2d -n openshift-operators |grep Image:
    Image:         quay.io/jbusche/pr187:rc2
```
And it's looking good - able to deploy samples and not seeing any error:
```
oc get pods -n default
NAME                                                              READY   STATUS      RESTARTS   AGE
jobtest-head-bb4ls                                                1/1     Running     0          5m25s
jobtest-worker-small-group-jobtest-ghfpq                          1/1     Running     0          5m25s
jobtest-worker-small-group-jobtest-zs52h                          1/1     Running     0          5m24s
```

**Comment by jbusche** [comment]
Please hold:
Tried another deploy:
- Made the image and pushed it to quay
- built the operator with my manual image
- Deploying the codeflare-stack kfdef worked

I'm testing now - but seeing something goofy...  not sure if it's with my PR or if it's something in the latest code.

**Comment by jbusche** [comment]
OK - it looks good to me:
My Manual operator image built and deployed fine - looks just like the regular one
Deploying the sample appwrappers looked ok.
Deploying the performance test appwrappers looked ok (Need to make a tweak on the perf script, that'll be in another PR later for MCAD)
@sutaakar do you think this looks good to merge?

File: controllers/mcad_controller.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func (r *MCADReconciler) DeleteResource(params *MCADParams, template string, fns
 // +kubebuilder:rbac:groups=codeflare.codeflare.dev,resources=mcads,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=codeflare.codeflare.dev,resources=mcads/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=codeflare.codeflare.dev,resources=mcads/finalizers,verbs=update
-// +kubebuilder:rbac:groups=workload.codeflare.dev,resources=queuejobs;schedulingspecs;appwrappers;appwrappers/finalizers;appwrappers/status,verbs=get;list;watch;create;update;patch;delete;deletecollection
+// +kubebuilder:rbac:groups=workload.codeflare.dev,resources=xqueuejobs;queuejobs;schedulingspecs;appwrappers;appwrappers/finalizers;appwrappers/status;quotasubtrees,verbs=get;list;watch;create;update;patch;delete;deletecollection
```
Comment (astefanutti) [inline controllers/mcad_controller.go:105]: I think `xqueuejobs` CRD has been removed from recent MCAD versions.

**Review by astefanutti — COMMENTED** [review]
It seems there are some extra commits. Could you please rebase it properly?

**Comment by jbusche** [comment]
Sure - let me close this and restart - there have been a lot of code changes recently - apparently my rebase wasn't correct.

**Pull request closed without merging**